scripts/sct_maths.py

```diff
@@ -14,20 +14,21 @@
 
 import os
 import sys
+
 import numpy as np
 import argparse
+import pickle
+import gzip
+import matplotlib
+import matplotlib.pyplot as plt
 
-import spinalcordtoolbox as sct
 from spinalcordtoolbox.image import Image
 from spinalcordtoolbox.utils import Metavar, SmartFormatter
-import spinalcordtoolbox.math
+import spinalcordtoolbox.math as math
```

Comment:
the reason i was specifically calling `sct.math` is to avoid confusion with [Python's math libary](https://docs.python.org/3/library/math.html). I was not sure what was good practice here…?

Reply:
Good point. We could import as `sct_math` to avoid potential clash.

Reply:
out of curiosity, why would `sct_math` be preferred over `sct.math`? (same number of chars)

Reply:
Do you mean `import spinalcordtoolbox.math as sct.math`? That is invalid syntax. 

It is a good practice to import the narrowest scope we need, therefore avoid importing the whole `spinalcordtoolbox` namespace.

Reply:
i actually meant 
~~~
import spinalcordtoolbox as sct
import spinalcordtoolbox.math

…
# when used:
sct.math
~~~

but our 2nd sentence (narrow scope) answers it

Reply:
> Good point. We could import as `sct_math` to avoid potential clash.

+1 to this suggestion. Even if we don't use Python's `math` module, I think it would still help with readability, so that it's clearer to someone skimming through the code that the functions come from SCT's `math.py`.

> It's even better to define a `__all__` in `math.py`. 

Could you clarify how `__all__` is relevant here? My understanding was that it's only for when `from <module> import *` is used, but we're not doing that sort of importing here, I don't think.

Reply:
> Could you clarify how __all__ is relevant here?

On the topic of narrowing the scope of imported variables. Using `__all__` allows to further narrow down what is imported.

> My understanding was that it's only for when from <module> import * is used, but we're not doing that sort of importing here, I don't think.

Yes you are right. I am proposing to add the `__all__` in API modules, which makes it cleaner for API consumers who use `from spinalcordtoolbox.math import *`. In this specific case it would have no effect, but it could be used by someones else who wants to build on top of the `math` API.

For the case of `math` it's not extremely relevant, because I only see 2 or 3 private variables in the module, but some other modules have a lot more. It does come with the cost of maintaining the `__all__` list.

I don't have a very strong opinion on this, but I think it would be a good practice to adopt.

Reply:
Ah, thank you, I understand better now.

Because this idea could apply to other modules too (and not just `sct.math`), would this discussion be better for a separate issue?

Reply:
> would this discussion be better for a separate issue?

Yes. I'm thinking of proposing changes in https://github.com/neuropoly/spinalcordtoolbox/blob/master/CONTRIBUTING.rst  and opening a PR to discuss/collaborate (wiki doesn't support PR which makes it not practical).

Reply:
Ah, sorry, I missed this question until just now. Hm... I wonder if this is something we could discuss on Slack? e.g. proposing a certain style or convention in #sct_dev, and if there's agreement, then just adding it directly to the Wiki.

Reply:
When it comes to simpler things like this (`__all__` seems simple to me, at least), I imagine we don't need a big process for it, just some informal chats between me/you/Chris/Nick/Julien instead.

Reply:
+1 for the chat